File: lib/ts/src/TelepactSchemaFiles.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { FsModule, PathModule } from './fileSystem.js';
import { getSchemaFileMap } from './internal/schema/GetSchemaFileMap.js';

export class TelepactSchemaFiles {
    filenamesToJson: Record<string, string>;

    constructor(directory: string, fs: FsModule, path: PathModule) {
        this.filenamesToJson = getSchemaFileMap(directory, fs, path);
    }
}

File: lib/go/TelepactSchemaFiles.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package telepact

import "github.com/telepact/telepact/lib/go/internal/schema"

// TelepactSchemaFiles holds the raw JSON contents of Telepact schema files keyed by filename.
type TelepactSchemaFiles struct {
	FilenamesToJSON map[string]string
}

// NewTelepactSchemaFiles constructs a TelepactSchemaFiles instance by reading schema files from a directory.
func NewTelepactSchemaFiles(directory string) (*TelepactSchemaFiles, error) {
	fileMap, parseFailures, err := schema.GetSchemaFileMap(directory)
	if err != nil {
		return nil, err
	}
	if len(parseFailures) > 0 {
		return nil, NewTelepactSchemaParseError(parseFailures, fileMap)
	}

	return &TelepactSchemaFiles{
		FilenamesToJSON: fileMap,
	}, nil
}
